Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,41 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find old name and old name symlinks for formula and remove them
+          begin
+            formula = Formulary.from_rack(keg.rack)
```

Comment:
I need formula to get `oldname`, so I either look into `FORMULA_RENAMES` and `TAP_FORMULA_RENAMES` or instantiate a formula or make `INSTALL_RECEIPT` store the oldname.